rotkehlchen/api/rest.py

```diff
@@ -4197,3 +4201,61 @@ def get_binance_savings_history(
             query_filter=query_filter,
             value_filter=value_filter,
         )
+
+    def get_ens_avatar(self, ens_name: str, match_header: Optional[str]) -> Response:
+        """
+        Searches for ENS avatar of the given `ens_name`.
+        If found returns the avatar, otherwise returns 404 status code.
+
+        Also supports etag mechanism that helps with caching on the API caller side.
+        """
+        avatar_path = self.rotkehlchen.data_dir / 'avatars' / f'{ens_name}.png'
```

Comment:
Since there's a `DBEns` class, why not move this logic into that class?
That way you'd only have to return the bytes data of the image.

Also, the return of an image is also repeated twice...
I'm proposing something like...
```python
return api_image_response(image_retrieval_func)
```

The `image_retrieval_func` can return `tuple[Optional[bytes], str]` where the `str` is the error message.
You might have better ideas here

Reply:
About DBEns -- I would not put the logic for reading a file there since I think it will be good to keep only db-related logic there.

As for `api_image_response` -- yeah, I think on how to abstract it.